### ISSUE TITLE
Better/earlier error handling of unexpected casing in 'conanfile.py'

### ISSUE THREAD

**Issue by davidkjaer** [conan-io/conan#1257]
Hi,

I have recently created a very simple conan package that I wanted to upload to our local conan server. I was using the conan package tools to run the process.

The project built, tested and packaged fine but when it was time to upload it failed with the following error: 
"ERROR: Cannot upload corrupted recipe [my name]"

After doing the ceremonial google search I tried to fiddle around with the package recipe in various ways. That didn't help either.

As a very last restort I downloaded the conan source code from github and found the culprit:
(conans/client/remote_manager.py l. 37)
``` python
        if CONANFILE not in files or CONAN_MANIFEST not in files:
            raise ConanException("Cannot upload corrupted recipe '%s'" % str(conan_reference))
```
It turns out that I had accidentally named my conan file 'conanFile.py' whereas the check above expects 'conanfile.py'. For some reason - probably because I was developing on Windows - everything leading up to upload worked just fine.

It would be very nice if the process were to fail earlier in this case. This would probably break backwards compatibility, though. If that fix is not doable a more telling error message would really have helped me.

**Comment by memsharded** [comment]
Certainly, the non case sensitiveness of Win is really annoying. Conan have some checks for packages, but not for the conanfile file itself. Agree, can be added. Next 0.23 is packed, I suggest for the next 0.24, not very high priority. 

Thanks for reporting!

**Comment by memsharded** [comment]
Fix released in 0.24, "Conanfile.py" shouldn't be a problem anymore :)
